src/test/scala/org/scalastyle/DirectoryTest.scala

```diff
@@ -0,0 +1,39 @@
+package org.scalastyle
+
+import java.io.File
+
+import org.junit.{Assert, Test}
+import org.scalatestplus.junit.AssertionsForJUnit
+
+class DirectoryTest extends AssertionsForJUnit {
+
+  private def toFile(s: String) = {
+    val base = "src/test/resources/testDir/"
+    new File(base + s)
+  }
+
+  @Test
+  def processSingleFileOnlyOnce(): Unit = {
+    val duplicatedFile = List("dirA/X.scala", "dirA/X.scala")
+    assertGetFiles(1, duplicatedFile)
+  }
+
+  @Test
+  def processDirectoryOnlyOnce(): Unit = {
+    val duplicatedDir = List("dirA", "dirA")
+    assertGetFiles(1, duplicatedDir)
+  }
+
+  @Test
+  def processFileInProcessDirOnlyOnce(): Unit = {
+    val duplicatedFileInDir = List("dirB", "dirB/Y.scala")
+    assertGetFiles(2, duplicatedFileInDir)
```

Comment:
Shouldn't this give you only one file back (excluding the root directory) since your logic [here](https://github.com/beautiful-scala/scalastyle/pull/1/files#diff-74cefba7e316a3be4cde4bb6b1278a5cR73) excludes directories?

Reply:
No, the directory is actually processed: its files are listed and added to the files to be processed ('currentFiles'), which is then done recursively.

And thank you for the fast response - it's great that someone is maintaining scalastyle again!

Reply:
Ok, I checked the code and now I understand what's happening here. Thanks again, will release it as a patch.